deep_quoridor/src/agents/alphazero/nn_evaluator.py

```diff
@@ -33,17 +33,16 @@ def __init__(
             self.rotated_action_mapping[rotated_action_i] = action_i
 
     def evaluate(self, game: Quoridor):
+        # Rotate the board if player 2 is playing so that we always work with player 1's perspective.
+        game, is_board_rotated = self.rotate_game_for_player_two(game)
+
         with torch.no_grad():
-            input_array, is_board_rotated = self.game_to_input_array(game)
+            input_array = self.game_to_input_array(game)
             unmasked_policy, value = self.network(torch.from_numpy(input_array).float().to(self.device))
             unmasked_policy = unmasked_policy.cpu().numpy()
             value = value.item()
 
-        if is_board_rotated:
-            unmasked_policy = unmasked_policy[self.rotated_action_mapping]
-            # TODO: Also rotate the action mask!
-
-        # Mask the policy to ignore invalid actions
+        # Mask the policy to ignore invalid actions. NOTE: Game is already rotated so the valid actions will be rotated too
```

Comment:
I like your approach of rotating the game first, but since rotate_board changes the game in place, and python passes everything by reference, it could be confusing to callers of this function that their game changed under them.

Reply:
But the returned rotated game is a deep_copy of the original one, it's not the originally changed one.
What do you suggest instead?